lm75bd/lm75bd.c

```diff
@@ -28,8 +28,29 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
-  return ERR_CODE_SUCCESS;
+    if(temp == NULL){
+         return ERR_CODE_INVALID_ARG;
+    }
+
+    uint8_t reg = TEMP_REG;
+    uint8_t tempDataRead[2];
+
+    i2cSendTo(devAddr, &reg, 1);
+
+    // Reading the temperature reg
+    i2cReceiveFrom(devAddr, tempDataRead, 2);
+    int16_t tempRaw = (uint16_t)(tempDataRead[0] << 3) | (uint16_t)(tempDataRead[1]>>5);
+
+    if ((tempDataRead[0] &( 1 << 7)) == 0) {
+        *temp = tempRaw * 0.125;
+    } else {
+        tempRaw = ~tempRaw + 1;
+        int16_t mask = 0xF800;
```

Comment:
this mask can just be a macro and I would add a comment explaining why it's that value